### ISSUE TITLE
No X apps support through Xwayland on Ubuntu 20.04

### ISSUE THREAD

**Issue by B1Z0N** [Llandy3d/sway-on-ubuntu#6]
When I try to run any X app in Sway it outputs: `Missing X server or $DISPLAY`. When I'm running Xwayland it prints that sway is missing wl-shell support. In [this](https://github.com/swaywm/sway/issues/1692) issue they dropped support for wl-shell. And [xorg-xserver](https://gitlab.freedesktop.org/xorg/xserver)(xwayland too in this repo) on latest commits uses xdg-shell.

Does someone have similar problems? Can't run on ubuntu20.04, probably need to update it to 22.04, but I can't, it's not my machine. Just in case, [here's](https://transfer.sh/kLqryQ/sway.log) my logs from running google chrome on sway.

**Comment by B1Z0N** [comment]
Solution: one should manually enable Xwayland on build to run apps like Chrome and other. In steps `Build wlroots` and `Build sway` replace `meson build` with `meson build -Dxwayland=enable`.


@Llandy3d may I open a PR to fix this? We should definitely enable wayland, lots of apps use it!

**Comment by Llandy3d** [comment]
Absolutely! It would be really appreciated, I will try to reproduce the issue and test the PR fixes as soon as I can get my hands on another machine for testing!

**Comment by Llandy3d** [comment]
alright, I got a vm to work and I was able to test the installation.
Following the current steps I seem to have x apps working correctly out of the box, I confirmed it with `xeyes` and `Chromium`.

Do you by any chance remember if you've installed these dependencies before building `wlroots` ?
```
sudo apt install libxcb-composite0-dev \
        libxcb-icccm4-dev \
        libxcb-image0-dev \
        libxcb-render0-dev \
        libxcb-xfixes0-dev \
        libxkbcommon-dev \
        libxcb-xinput-dev \
        libx11-xcb-dev
```
